crates/ruff_linter/src/rules/ruff/rules/unnecessary_dict_comprehension_for_iterable.rs

```diff
@@ -0,0 +1,192 @@
+use anyhow::{bail, Result};
+use libcst_native::{Arg, Call, Expression, Name, ParenthesizableWhitespace};
+use ruff_diagnostics::{AlwaysFixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{self as ast, Comprehension, Expr};
+use ruff_python_codegen::Stylist;
+use ruff_source_file::Locator;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::cst::matchers::match_expression;
+use crate::fix::codemods::CodegenStylist;
+use crate::fix::edits::pad;
+
+/// ## What it does
+/// Checks for unnecessary `dict` comprehension when creating a new dictionary from iterable.
+///
+/// ## Why is this bad?
+/// It's unnecessary to use a `dict` comprehension to build a dictionary from an iterable when the value is `static`.
+/// Use `dict.fromkeys(iterable)` instead of `{value: None for value in iterable}`.
+///
+/// `dict.fromkeys(iterable)` is more readable and faster than `{value: None for value in iterable}`.
+///
+///
+/// ## Examples
+/// ```python
+/// {a: None for a in iterable}
+/// {a: 1 for a in iterable}
+/// ```
+///
+/// Use instead:
+/// ```python
+/// dict.fromkeys(iterable)
+/// dict.fromkeys(iterable, 1)
+/// ```
+#[violation]
+pub struct UnnecessaryDictComprehensionForIterable {
+    is_value_none_literal: bool,
```

Comment:
nit: we could use an enum instead of a boolean. A boolean value is hard to read until it is linked with some context like a variable name. An enum can prove to be readable in such cases.
```rust
enum Value {
	// A `None` literal
	NoneLiteral,
	// Any value other than `None`
	Other
}
```

Reply:
Yea, I thought about using an Enum but it kind of felt slightly over board for such a simple thing.